onnxruntime/test/providers/qnn/conv_test.cc

```diff
@@ -13,6 +13,73 @@
 namespace onnxruntime {
 namespace test {
 
+// The bug is from a QDQ model, and Conv node gets processed before it's producer Mul node
+// A Transpose node gets inserted between Mul and the dynamic weight tensor shape on Conv
+// to make Conv weight with shape HWNC
+// However it changes Mul output shape to HWNC and cause issue
+// It has to be QDQ model, because the DQ node with initializer on Conv gets processed first
+// and DQ node requires its node unit to be processed
+// So, Conv gets processed before Mul node
+TEST_F(QnnCPUBackendTests, Test_QDQConvWithDynamicWeightsFromMul) {
```

Comment:
Is this test still testing the issue addressed by this PR (wrong weight input name when the weight is dynamic)? Just want to make sure.

Reply:
Right, this test will be failed without this PR.

---
In reply to: [1223644268](https://github.com/microsoft/onnxruntime/pull/16235#discussion_r1223644268) [](ancestors = 1223644268)